### DISCUSSION TITLE
Debug PyScaffold extension created with PyScaffold in PyCharm

### DISCUSSION THREAD

**Discussion by maecky** [pyscaffold/pyscaffold#722] [Best Practices & Workflows]
I created my first PyScaffol extension like this:
```
putup first_scaff_extension --custom-extension --markdown --pre-commit 
python -m venv .venv
pre-commit autoupdate
pip install -e .
pip install -e .["testing"]
```

I am able to build my extension with tox and use it but I would like to debug the extension with pyCharm. How can I setup a convenient development workflow for this?

Thanks in advance!

**Comment by abravalheri** [comment]
Hi @maecky, I am not a PyCharm user, so the amount of help I can provide you is very limited...
By having a look on the PyCharm docs, I believe that you can do the following:

1. Configure PyCharm to use the virtual environment you have previously created for your project: https://www.jetbrains.com/help/pycharm/creating-virtual-environment.html
   See the "Existing virtual environment" section in the docs
2. Configure PyCharm to use `tox` for running the tests: https://www.jetbrains.com/help/pycharm/run-debug-configuration-tox.html

You can also configure PyCharm with a custom run command to run whichever `tox` environment you need: https://www.jetbrains.com/help/pycharm/run-debug-configuration.html#run-debug-parameters
